File: android_client_app/app/src/main/java/io/orbitlogic/client/ui/home/HomeScreen.kt
package io.orbitlogic.client.ui.home

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import io.orbitlogic.client.data.models.PackageInfo
import io.orbitlogic.client.data.models.UserProfile
import io.orbitlogic.client.ui.theme.*

@Composable
fun HomeScreen(
    user: UserProfile,
    packages: List<PackageInfo>,
    onActionSelected: (String) -> Unit
) {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(DarkBackground)
            .padding(16.dp)
    ) {
        // Custom Greeting Header
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Row(verticalAlignment = Alignment.CenterVertically) {
                Box(
                    modifier = Modifier
                        .size(40.dp)
                        .background(ClientCyan, CircleShape),
                    contentAlignment = Alignment.Center
                ) {
                    Text(
                        text = if (user.name.isNotEmpty()) user.name[0].uppercase() else "U",
                        fontWeight = FontWeight.Bold,
                        color = Color.White
                    )
                }
                Spacer(modifier = Modifier.width(12.dp))
                Column {
                    Text("Good morning", color = TextSecondary, fontSize = 11.sp)
                    Text("Hi, ${user.name.split(" ")[0]}", color = Color.White, fontSize = 16.sp, fontWeight = FontWeight.W900)
                }
            }
            Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                IconButton(onClick = {}) {
                    Icon(imageVector = Icons.Default.Search, contentDescription = null, tint = Color.White)
                }
                IconButton(onClick = {}) {
                    Icon(imageVector = Icons.Default.Notifications, contentDescription = null, tint = Color.White)
                }
            }
        }
        Spacer(modifier = Modifier.height(20.dp))

        // 2x2 Grid Actions
        Text("Quick Access", color = Color.White, fontSize = 15.sp, fontWeight = FontWeight.Bold)
        Spacer(modifier = Modifier.height(10.dp))
        Column(verticalArrangement = Arrangement.spacedBy(10.dp)) {
            Row(horizontalArrangement = Arrangement.spacedBy(10.dp), modifier = Modifier.fillMaxWidth()) {
                GridCard(
                    title = "Book Now",
                    desc = "Schedule a session",
                    icon = Icons.Default.CalendarToday,
                    color = ClientCyan,
                    modifier = Modifier.weight(1f)
                ) { onActionSelected("booking") }
                GridCard(
                    title = "Track Order",
                    desc = "Live shooter updates",
                    icon = Icons.Default.Radar,
                    color = ClientPurple,
                    modifier = Modifier.weight(1f)
                ) { onActionSelected("tracking") }
            }
            Row(horizontalArrangement = Arrangement.spacedBy(10.dp), modifier = Modifier.fillMaxWidth()) {
                GridCard(
                    title = "Packages",
                    desc = "Check pricing options",
                    icon = Icons.Default.Inventory,
                    color = Color(0xFF2D6A4F),
                    modifier = Modifier.weight(1f)
                ) { onActionSelected("packages") }
                GridCard(
                    title = "Brand DNA",
                    desc = "Style presets",
                    icon = Icons.Default.AutoAwesome,
                    color = Color(0xFFFFB300),
                    modifier = Modifier.weight(1f)
                ) { onActionSelected("brand_dna") }
            }
        }

        Spacer(modifier = Modifier.height(24.dp))

        // Packages horizontal row
        Text("Our Packages", color = Color.White, fontSize = 15.sp, fontWeight = FontWeight.Bold)
        Spacer(modifier = Modifier.height(10.dp))
        LazyRow(horizontalArrangement = Arrangement.spacedBy(12.dp)) {
            items(packages) { pkg ->
                PackagePreviewCard(pkg)
            }
        }

        Spacer(modifier = Modifier.height(24.dp))

        // Large Promo CTA Card
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .background(
                    brush = Brush.linearGradient(
                        listOf(ClientCyan.copy(alpha = 0.15f), ClientPurple.copy(alpha = 0.15f))
                    ),
                    shape = RoundedCornerShape(16.dp)
                )
                .border(1.dp, BorderColor, RoundedCornerShape(16.dp))
                .padding(20.dp)
        ) {
            Column {
                Text(
                    text = "Ready to Create\nSomething Cinematic?",
                    fontSize = 18.sp,
                    fontWeight = FontWeight.W900,
                    color = Color.White,
                    lineHeight = 22.sp
                )
                Spacer(modifier = Modifier.height(6.dp))
                Text(
                    text = "Professional cinematic edits delivered directly to your device within 60 minutes.",
                    fontSize = 11.sp,
                    color = TextSecondary
                )
                Spacer(modifier = Modifier.height(16.dp))
                Button(
                    onClick = { onActionSelected("booking") },
                    colors = ButtonDefaults.buttonColors(containerColor = Color.White)
                ) {
                    Text("Book Session", color = Color.Black, fontWeight = FontWeight.Bold)
                }
            }
        }
    }
}

@Composable
fun GridCard(
    title: String,
    desc: String,
    icon: ImageVector,
    color: Color,
    modifier: Modifier = Modifier,
    onClick: () -> Unit
) {
    Box(
        modifier = modifier
            .height(100.dp)
            .background(DarkSurface, RoundedCornerShape(16.dp))
            .border(1.dp, BorderColor, RoundedCornerShape(16.dp))
            .clickable { onClick() }
            .padding(16.dp)
    ) {
        Column(verticalArrangement = Arrangement.SpaceBetween, modifier = Modifier.fillMaxSize()) {
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(title, color = Color.White, fontSize = 14.sp, fontWeight = FontWeight.W900)
                Box(
                    modifier = Modifier
                        .size(28.dp)
                        .background(color.copy(alpha = 0.15f), RoundedCornerShape(8.dp)),
                    contentAlignment = Alignment.Center
                ) {
                    Icon(imageVector = icon, contentDescription = null, tint = color, modifier = Modifier.size(16.dp))
                }
            }
            Text(desc, color = TextSecondary, fontSize = 10.sp)
        }
    }
}

@Composable
fun PackagePreviewCard(pkg: PackageInfo) {
    Box(
        modifier = Modifier
            .width(180.dp)
            .background(DarkSurface, RoundedCornerShape(16.dp))
            .border(1.dp, BorderColor, RoundedCornerShape(16.dp))
            .padding(16.dp)
    ) {
        Column {
            Text(pkg.name, color = Color.White, fontSize = 13.sp, fontWeight = FontWeight.Bold)
            Spacer(modifier = Modifier.height(4.dp))
            Text(pkg.focus, color = TextSecondary, fontSize = 10.sp)
            Spacer(modifier = Modifier.height(16.dp))
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text("₹${pkg.price}", color = ClientCyan, fontSize = 14.sp, fontWeight = FontWeight.W900)
                Text(pkg.deliveryTime, color = TextSecondary, fontSize = 9.sp)
            }
        }
    }
}

File: client_app/lib/features/home/home_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:go_router/go_router.dart';
import '../../core/theme.dart';
import '../../core/api_service.dart';
import '../../core/storage_service.dart';
import '../../models/user.dart';
import '../../models/booking.dart';
import '../booking/booking_flow_screen.dart';
import '../tracking/tracking_view.dart';

final userProfileProvider = StateProvider<UserProfile?>((ref) => null);
final bookingsListProvider = StateProvider<List<BookingInfo>>((ref) => []);

class HomeScreen extends ConsumerStatefulWidget {
  const HomeScreen({super.key});

  @override
  ConsumerState<HomeScreen> createState() => _HomeScreenState();
}

class _HomeScreenState extends ConsumerState<HomeScreen> {
  int _currentIndex = 0;

  @override
  void initState() {
    super.initState();
    _loadProfileAndBookings();
  }

  void _loadProfileAndBookings() async {
    final storage = StorageService();
    final profile = await storage.getUser();
    if (profile != null && mounted) {
      ref.read(userProfileProvider.notifier).state = profile;
      
      final api = ApiService();
      final bookings = await api.fetchClientBookings(profile.id);
      ref.read(bookingsListProvider.notifier).state = bookings;
    }
  }

  void _logout() async {
    final storage = StorageService();
    await storage.clear();
    if (mounted) context.go('/login');
  }

  @override
  Widget build(BuildContext context) {
    final user = ref.watch(userProfileProvider);
    if (user == null) {
      return const Scaffold(body: Center(child: CircularProgressIndicator()));
    }

    final views = [
      _HomeFeed(user: user, onNavigate: (index) => setState(() => _currentIndex = index)),
      const _PackagesView(),
      const TrackingView(),
      _ProfileView(user: user, onLogout: _logout),
    ];

    return Scaffold(
      body: SafeArea(
        child: Column(
          children: [
            // Custom Header Navbar (matches design images!)
            Padding(
              padding: const EdgeInsets.symmetric(horizontal: 16.0, vertical: 12.0),
              child: Row(
                children: [
                  // Avatar with initials
                  CircleAvatar(
                    radius: 20,
                    backgroundColor: OrbitTheme.clientCyan,
                    child: Text(
                      user.name.isNotEmpty ? user.name[0].toUpperCase() : 'U',
                      style: const TextStyle(fontWeight: FontWeight.w900, color: Colors.white, fontSize: 14),
                    ),
                  ),
                  const SizedBox(width: 12),
                  // Greetings
                  Expanded(
                    child: Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        const Text('Good morning', style: TextStyle(color: OrbitTheme.textSecondary, fontSize: 11)),
                        Text(
                          'Hi, ${user.name.split(' ')[0]}',
                          style: const TextStyle(fontSize: 16, fontWeight: FontWeight.w900, color: Colors.white),
                        ),
                      ],
                    ),
                  ),
                  // Actions buttons row
                  Row(
                    children: [
                      _NavIconBtn(icon: Icons.search, onTap: () {}),
                      const SizedBox(width: 8),
                      _NavIconBtn(icon: Icons.notifications, onTap: () {}),
                      const SizedBox(width: 8),
                      _NavIconBtn(
                        icon: Icons.keyboard_arrow_down,
                        onTap: () {
                          // Dropdown overlay
                          showModalBottomSheet(
                            context: context,
                            backgroundColor: OrbitTheme.cardBackground,
                            shape: const RoundedRectangleBorder(borderRadius: BorderRadius.vertical(top: Radius.circular(20))),
                            builder: (context) => Container(
                              padding: const EdgeInsets.all(24),
                              child: Column(
                                mainAxisSize: MainAxisSize.min,
                                children: [
                                  ListTile(
                                    leading: const Icon(Icons.person, color: OrbitTheme.clientCyan),
                                    title: const Text('My Profile', style: TextStyle(fontWeight: FontWeight.bold)),
                                    onTap: () {
                                      Navigator.pop(context);
                                      setState(() => _currentIndex = 3);
                                    },
                                  ),
                                  const Divider(color: OrbitTheme.border),
                                  ListTile(
                                    leading: const Icon(Icons.logout, color: Colors.redAccent),
                                    title: const Text('Log Out', style: TextStyle(color: Colors.redAccent, fontWeight: FontWeight.bold)),
                                    onTap: () {
                                      Navigator.pop(context);
                                      _logout();
                                    },
                                  ),
                                ],
                              ),
                            ),
                          );
                        },
                      ),
                    ],
                  ),
                ],
              ),
            ),
            
            // Subtitle
            const Padding(
              padding: EdgeInsets.symmetric(horizontal: 16.0),
              child: Align(
                alignment: Alignment.centerLeft,
                child: Text(
                  'Ready to create something cinematic?',
                  style: TextStyle(color: OrbitTheme.textSecondary, fontSize: 13),
                ),
              ),
            ),
            const SizedBox(height: 12),

            Expanded(child: views[_currentIndex]),
          ],
        ),
      ),
      bottomNavigationBar: Container(
        padding: const EdgeInsets.only(bottom: 12, left: 16, right: 16, top: 4),
        color: Colors.transparent,
        child: Container(
          height: 64,
          decoration: BoxDecoration(
            color: const Color(0xCC050505),
            borderRadius: BorderRadius.circular(28),
            border: Border.all(color: OrbitTheme.clientCyan.withOpacity(0.12)),
          ),
          child: Row(
            mainAxisAlignment: MainAxisAlignment.spaceAround,
            children: [
              _BottomNavItem(
                icon: Icons.home_outlined,
                activeIcon: Icons.home,
                label: 'Home',
                isActive: _currentIndex == 0,
                onTap: () => setState(() => _currentIndex = 0),
              ),
              _BottomNavItem(
                icon: Icons.wallet_giftcard_outlined,
                activeIcon: Icons.wallet_giftcard,
                label: 'Packages',
                isActive: _currentIndex == 1,
                onTap: () => setState(() => _currentIndex = 1),
              ),
              _BottomNavItem(
                icon: Icons.radar,
                activeIcon: Icons.radar,
                label: 'Track',
                isActive: _currentIndex == 2,
                onTap: () => setState(() => _currentIndex = 2),
              ),
              _BottomNavItem(
                icon: Icons.person_outline,
                activeIcon: Icons.person,
                label: 'Profile',
                isActive: _currentIndex == 3,
                onTap: () => setState(() => _currentIndex = 3),
              ),
            ],
          ),
        ),
      ),
    );
  }
}

class _NavIconBtn extends StatelessWidget {
  final IconData icon;
  final VoidCallback onTap;

  const _NavIconBtn({required this.icon, required this.onTap});

  @override
  Widget build(BuildContext context) {
    return GestureDetector(
      onTap: onTap,
      child: Container(
        width: 38,
        height: 38,
        decoration: BoxDecoration(
          color: Colors.white.withOpacity(0.08),
          shape: BoxShape.circle,
        ),
        child: Icon(icon, color: Colors.white, size: 20),
      ),
    );
  }
}

class _BottomNavItem extends StatelessWidget {
  final IconData icon;
  final IconData activeIcon;
  final String label;
  final bool isActive;
  final VoidCallback onTap;

  const _BottomNavItem({
    required this.icon,
    required this.activeIcon,
    required this.label,
    required this.isActive,
    required this.onTap,
  });

  @override
  Widget build(BuildContext context) {
    return GestureDetector(
      onTap: onTap,
      behavior: HitTestBehavior.opaque,
      child: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          Icon(
            isActive ? activeIcon : icon,
            color: isActive ? OrbitTheme.clientCyan : OrbitTheme.textSecondary.withOpacity(0.5),
            size: 22,
          ),
          const SizedBox(height: 3),
          Text(
            label,
            style: TextStyle(
              fontSize: 9,
              color: isActive ? OrbitTheme.clientCyan : OrbitTheme.textSecondary.withOpacity(0.5),
              fontWeight: isActive ? FontWeight.bold : FontWeight.normal,
            ),
          ),
        ],
      ),
    );
  }
}

// ─── Home Feed ────────────────────────────────────────────────────────────────
class _HomeFeed extends ConsumerWidget {
  final UserProfile user;
  final Function(int) onNavigate;

  const _HomeFeed({required this.user, required this.onNavigate});

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final bookings = ref.watch(bookingsListProvider);
    final hasActive = bookings.any((b) => b.status != 'DELIVERED' && b.status != 'CANCELLED');

    return ListView(
      padding: const EdgeInsets.all(16.0),
      children: [
        // 2x2 Grid Actions
        GridView.count(
          crossAxisCount: 2,
          shrinkWrap: true,
          physics: const NeverScrollableScrollPhysics(),
          mainAxisSpacing: 12,
          crossAxisSpacing: 12,
          childAspectRatio: 1.25,
          children: [
            _GridCard(
              icon: Icons.calendar_today,
              color: const Color(0xFF00BFFF),
              title: 'Book Now',
              desc: 'Schedule a session',
              onTap: () => context.push('/booking'),
            ),
            _GridCard(
              icon: Icons.radar,
              color: const Color(0xFFA020F0),
              title: 'Track Order',
              desc: hasActive ? 'Active Shoot' : 'No active',
              onTap: () => onNavigate(2),
            ),
            _GridCard(
              icon: Icons.card_giftcard,
              color: const Color(0xFF2D6A4F),
              title: 'Packages',
              desc: 'View pricing',
              onTap: () => onNavigate(1),
            ),
            _GridCard(
              icon: Icons.bolt,
              color: const Color(0xFFFFB300),
              title: 'Brand DNA',
              desc: 'Customize style',
              onTap: () => onNavigate(1),
            ),
          ],
        ),
        
        const SizedBox(height: 28),
        
        // Horizontal list packages preview
        const Row(
          mainAxisAlignment: MainAxisAlignment.spaceBetween,
          children: [
            Text('Our Packages', style: TextStyle(fontSize: 16, fontWeight: FontWeight.bold)),
            Text('View All >', style: TextStyle(color: OrbitTheme.clientCyan, fontSize: 12)),
          ],
        ),
        const SizedBox(height: 12),
        SizedBox(
          height: 140,
          child: ListView(
            scrollDirection: Axis.horizontal,
            children: [
              _PkgItemCard(
                title: 'Personalized',
                price: '₹1,999/session',
                bullet: '1 Cinematic Reel',
                color: OrbitTheme.clientCyan.withOpacity(0.05),
              ),
              const SizedBox(width: 12),
              _PkgItemCard(
                title: 'Professional (UGC)',
                price: '₹4,999/session',
                bullet: '3 Cinematic Reels',
                color: OrbitTheme.clientPurple.withOpacity(0.05),
                isPopular: true,
              ),
            ],
          ),
        ),

        const SizedBox(height: 28),

        // Statistics bar
        Container(
          padding: const EdgeInsets.symmetric(vertical: 16),
          decoration: BoxDecoration(
            color: OrbitTheme.cardBackground,
            borderRadius: BorderRadius.circular(16),
            border: Border.all(color: OrbitTheme.border),
          ),
          child: const Row(
            mainAxisAlignment: MainAxisAlignment.spaceAround,
            children: [
              Column(
                children: [
                  Text('60 min', style: const TextStyle(fontWeight: FontWeight.w900, fontSize: 16, color: OrbitTheme.clientCyan)),
                  SizedBox(height: 4),
                  Text('Delivery Guarantee', style: TextStyle(fontSize: 10, color: OrbitTheme.textSecondary)),
                ],
              ),
              Column(
                children: [
                  Text('4K HDR', style: const TextStyle(fontWeight: FontWeight.w900, fontSize: 16, color: OrbitTheme.clientCyan)),
                  SizedBox(height: 4),
                  Text('Native Quality', style: TextStyle(fontSize: 10, color: OrbitTheme.textSecondary)),
                ],
              ),
              Column(
                children: [
                  Text('500+', style: const TextStyle(fontWeight: FontWeight.w900, fontSize: 16, color: OrbitTheme.clientCyan)),
                  SizedBox(height: 4),
                  Text('Reels Delivered', style: TextStyle(fontSize: 10, color: OrbitTheme.textSecondary)),
                ],
              ),
            ],
          ),
        ),
        
        const SizedBox(height: 24),
        
        // Large Promo CTA Card
        Container(
          padding: const EdgeInsets.all(24),
          decoration: BoxDecoration(
            gradient: LinearGradient(
              colors: [OrbitTheme.clientCyan.withOpacity(0.2), OrbitTheme.clientPurple.withOpacity(0.2)],
            ),
            borderRadius: BorderRadius.circular(20),
            border: Border.all(color: OrbitTheme.clientCyan.withOpacity(0.2)),
          ),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              const Text(
                'Ready to Create\nSomething Cinematic?',
                style: TextStyle(fontSize: 20, fontWeight: FontWeight.w900, height: 1.2),
              ),
              const SizedBox(height: 8),
              const Text(
                'Professional cinematic edits delivered directly to your device within 60 minutes.',
                style: TextStyle(fontSize: 12, color: OrbitTheme.textSecondary),
              ),
              const SizedBox(height: 20),
              ElevatedButton.icon(
                style: ElevatedButton.styleFrom(
                  backgroundColor: Colors.white,
                  foregroundColor: Colors.black,
                  shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(12)),
                  padding: const EdgeInsets.symmetric(horizontal: 20, vertical: 12),
                ),
                icon: const Icon(Icons.bolt, color: OrbitTheme.clientCyan),
                label: const Text('Book a Session Now', style: TextStyle(fontWeight: FontWeight.bold)),
                onPressed: () => context.push('/booking'),
              ),
            ],
          ),
        ),
        const SizedBox(height: 24),
      ],
    );
  }
}

class _GridCard extends StatelessWidget {
  final IconData icon;
  final Color color;
  final String title;
  final String desc;
  final VoidCallback onTap;

  const _GridCard({
    required this.icon,
    required this.color,
    required this.title,
    required this.desc,
    required this.onTap,
  });

  @override
  Widget build(BuildContext context) {
    return GestureDetector(
      onTap: onTap,
      child: Container(
        padding: const EdgeInsets.all(16),
        decoration: BoxDecoration(
          color: OrbitTheme.cardBackground,
          borderRadius: BorderRadius.circular(20),
          border: Border.all(color: OrbitTheme.border),
        ),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            Container(
              padding: const EdgeInsets.all(8),
              decoration: BoxDecoration(
                color: color.withOpacity(0.1),
                shape: BoxShape.circle,
              ),
              child: Icon(icon, color: color, size: 20),
            ),
            const SizedBox(height: 12),
            Text(title, style: const TextStyle(fontWeight: FontWeight.bold, fontSize: 13, color: Colors.white)),
            const SizedBox(height: 2),
            Text(desc, style: const TextStyle(color: OrbitTheme.textSecondary, fontSize: 10)),
          ],
        ),
      ),
    );
  }
}

class _PkgItemCard extends StatelessWidget {
  final String title;
  final String price;
  final String bullet;
  final Color color;
  final bool isPopular;

  const _PkgItemCard({
    required this.title,
    required this.price,
    required this.bullet,
    required this.color,
    this.isPopular = false,
  });

  @override
  Widget build(BuildContext context) {
    return Container(
      width: 180,
      padding: const EdgeInsets.all(16),
      decoration: BoxDecoration(
        color: color,
        borderRadius: BorderRadius.circular(16),
        border: Border.all(color: OrbitTheme.border),
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceBetween,
            children: [
              Text(title, style: const TextStyle(fontWeight: FontWeight.w900, fontSize: 13)),
              if (isPopular)
                Container(
                  padding: const EdgeInsets.symmetric(horizontal: 6, vertical: 2),
                  decoration: BoxDecoration(color: OrbitTheme.clientPurple, borderRadius: BorderRadius.circular(4)),
                  child: const Text('POPULAR', style: TextStyle(fontSize: 7, fontWeight: FontWeight.bold)),
                ),
            ],
          ),
          const SizedBox(height: 6),
          Text(price, style: const TextStyle(color: OrbitTheme.clientCyan, fontSize: 12, fontWeight: FontWeight.bold)),
          const SizedBox(height: 12),
          Row(
            children: [
              const Icon(Icons.check_circle_outline, color: Colors.greenAccent, size: 12),
              const SizedBox(width: 6),
              Text(bullet, style: const TextStyle(fontSize: 10, color: OrbitTheme.textSecondary)),
            ],
          ),
        ],
      ),
    );
  }
}

// ─── Placeholder Packages ─────────────────────────────────────────────────────
class _PackagesView extends StatelessWidget {
  const _PackagesView();

  @override
  Widget build(BuildContext context) {
    return const Center(child: Text('Packages Screen Details Here'));
  }
}

// ─── Placeholder Tracking ─────────────────────────────────────────────────────
class _TrackingView extends StatelessWidget {
  const _TrackingView();

  @override
  Widget build(BuildContext context) {
    return const Center(child: Text('Order Status Tracking Here'));
  }
}

// ─── Profile Tab ──────────────────────────────────────────────────────────────
class _ProfileView extends StatelessWidget {
  final UserProfile user;
  final VoidCallback onLogout;

  const _ProfileView({required this.user, required this.onLogout});

  @override
  Widget build(BuildContext context) {
    return ListView(
      padding: const EdgeInsets.all(16.0),
      children: [
        Container(
          padding: const EdgeInsets.all(16),
          decoration: BoxDecoration(
            color: OrbitTheme.cardBackground,
            borderRadius: BorderRadius.circular(16),
            border: Border.all(color: OrbitTheme.border),
          ),
          child: Column(
            children: [
              CircleAvatar(
                radius: 36,
                backgroundColor: OrbitTheme.clientCyan,
                child: Text(user.name.isNotEmpty ? user.name[0].toUpperCase() : 'U', style: const TextStyle(fontSize: 28, fontWeight: FontWeight.bold, color: Colors.white)),
              ),
              const SizedBox(height: 12),
              Text(user.name, style: const TextStyle(fontSize: 18, fontWeight: FontWeight.w900)),
              const SizedBox(height: 4),
              Text(user.email, style: const TextStyle(color: OrbitTheme.textSecondary, fontSize: 12)),
              if (user.phone.isNotEmpty) ...[
                const SizedBox(height: 4),
                Text(user.phone, style: const TextStyle(color: OrbitTheme.textSecondary, fontSize: 12)),
              ],
            ],
          ),
        ),
        const SizedBox(height: 24),
        ListTile(
          leading: const Icon(Icons.shield_outlined, color: OrbitTheme.clientCyan),
          title: const Text('Privacy Shield', style: TextStyle(fontSize: 13, fontWeight: FontWeight.bold)),
          trailing: const Icon(Icons.chevron_right, size: 16),
          onTap: () {},
        ),
        const Divider(color: OrbitTheme.border),
        ListTile(
          leading: const Icon(Icons.settings_outlined, color: OrbitTheme.clientCyan),
          title: const Text('App Settings', style: TextStyle(fontSize: 13, fontWeight: FontWeight.bold)),
          trailing: const Icon(Icons.chevron_right, size: 16),
          onTap: () {},
        ),
        const Divider(color: OrbitTheme.border),
        ListTile(
          leading: const Icon(Icons.logout, color: Colors.redAccent),
          title: const Text('Log Out', style: TextStyle(color: Colors.redAccent, fontSize: 13, fontWeight: FontWeight.bold)),
          trailing: const Icon(Icons.chevron_right, size: 16),
          onTap: onLogout,
        ),
      ],
    );
  }
}
